internal/cmd/plan.go

```diff
@@ -311,9 +311,19 @@ func PaymentMethodHelper(client *turso.Client, selected string) (bool, error) {
 	spinner := prompt.Spinner("Waiting for you to add a payment method")
 	defer spinner.Stop()
 
+	return checkPaymentMethod(client, "")
+}
+
+func checkPaymentMethod(client *turso.Client, stripeId string) (bool, error) {
 	errsInARoW := 0
+	var hasPaymentMethod bool
+	var err error
 	for {
-		hasPaymentMethod, err := client.Billing.HasPaymentMethod()
+		if stripeId != "" {
+			hasPaymentMethod, err = client.Billing.HasPaymentMethodWithStripeId(stripeId)
+		} else {
+			hasPaymentMethod, err = client.Billing.HasPaymentMethod()
+		}
```

Comment:
To avoid this, you can extract the conditional into a function:

```go
func hasPaymentMethod(client *turso.Client, stripeId string) (bool, error) {
	if stripeId != "" {
		return client.Billing.HasPaymentMethodWithStripeId(stripeId)
	}
	return client.Billing.HasPaymentMethod()
}
```